### PR TITLE
Create map to save & show ubication

### LINKED ISSUE #26: Create a map to save ubication
Objective map
 - click on every place and ubication will be saved 
 - we can search typing the ubication and the map will localize the direction

### PR DESCRIPTION
**Pull request by angel23v**
## Quick Info
   - Map to save the ubication where you click
   - Search button tool
   - Show map with the ubication saved when you edit an event
   - Navigate inside the map with the cursor
   - FIll automatically ubication fields on the form  (ubicacion, latitud, longitud)

Migrations? 
Yes, added latitud & longitud columns

## What does this change?
The way to save & show the ubication

## Why are you changing that?
Because it looks better with a interactive map and makes a better experience for the user 

## How do you manually test this?
Clicking many times on the map to see if the infomation was showed on the ubication fields.
Saving events and see the correct ubication info on my view
Editing events to see if the ubicatiion saved was showed on the map

## Screenshots
add_event
show_and_edit_ubi

### CODE REVIEW

File: app/javascript/controllers/map_controller.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
+import { Controller } from "@hotwired/stimulus";
+
+// Connects to data-controller="map"
+export default class extends Controller {
+  connect() {
+    let latitud = document.querySelector(".latitud");
+    let longitud = document.querySelector(".longitud");
+    let ubicacion = document.querySelector(".ubicacion");
+
+    const apiKey = "pk.8ef046976867e15790d27dae0c26d553";
```
Comment (MauricioNC) [inline app/javascript/controllers/map_controller.js]: Don't leave this api key directly in the code, please move it to an environment variable instead

File: app/views/layouts/_navbar.html.erb [hunk 2]
```diff
@@ -2,12 +2,16 @@
...
       <ul class="navbar-nav me-auto mb-2 mb-lg-0">
+          <% if !signed_in?%>
+            <p class= "text-light">SISTEMA DE GESTION DE EVENTOS</p> 
+          <% end %>
+        <li class="nav-item">
+          <% if signed_in?%>
+            <%= link_to "Sistema de Gestion Eventos", events_main_path , class: "btn btn-dark text-light"%>
+          <% end %>
+        </li>
         <li class="nav-item">
```
Comment (MauricioNC) [inline app/views/layouts/_navbar.html.erb]: You can improve the logic of this code. Have you used the unless statement?

File: app/views/layouts/application.html.erb [hunk 3]
```diff
@@ -5,14 +5,21 @@
     <meta name="viewport" content="width=device-width,initial-scale=1">
     <%= csrf_meta_tags %>
     <%= csp_meta_tag %>
-
+
```
Comment (MauricioNC) [inline app/views/layouts/application.html.erb]: Remove blank spaces

**Review by MauricioNC — CHANGES_REQUESTED** [review]

File: app/views/layouts/_navbar.html.erb (hunk 2 above)
Comment (angel23v) [reply app/views/layouts/_navbar.html.erb:15]: Yes, It will be better if I use ternary operator?

Comment (MauricioNC) [reply app/views/layouts/_navbar.html.erb]: No, ternary operator is recommended to use for short evaluations.

- In the first if, you are evaluating a negative condition, you can use unless instead.
- In the second if, you have the li tag outside of the if statement, if the condition is not met, you will have an empty li tag, so you can move the li tag inside the if statement, by this way, if the condition is not met, the li tag will not be created.

Comment (MauricioNC) [reply app/views/layouts/_navbar.html.erb]: or you can use an if else statement, but sometimes the linter marks a suggestion to replace the negative conditions to unless statement.

File: app/javascript/controllers/map_controller.js (hunk 1 above)
Comment (angel23v) [reply app/javascript/controllers/map_controller.js]: is there any documentation?

File: app/views/layouts/_navbar.html.erb (hunk 2 above)
Comment (angel23v) [reply app/views/layouts/_navbar.html.erb]: Thanks I already try with unless statement

File: app/javascript/controllers/map_controller.js (hunk 1 above)
Comment (MauricioNC) [reply app/javascript/controllers/map_controller.js]: Personally, I've never faced to this kind of situation. May be a solution could be save that api key into the credentials file and then create a method that returns the api key. 

To edit this file you can run `rails credentials:edit`.

And just call this method from Stimulus.

Comment (angel23v) [reply app/javascript/controllers/map_controller.js]: Thanks I try with that option & I wasn`t able to solve but searching I found a gem call "dotenv" and it works for me.

**Review by MauricioNC — APPROVED** [review]
LGTM!

**Review by juperameza — APPROVED** [review]
LGTM

**Pull request merged**